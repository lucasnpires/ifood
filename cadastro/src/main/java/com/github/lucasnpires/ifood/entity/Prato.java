package com.github.lucasnpires.ifood.entity;

import java.math.BigDecimal;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

@Entity
@Table(name = "prato")
public class Prato extends PanacheEntityBase{
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private String nome;
	
	private String descricao;
	
	@ManyToOne
	private Restaurante restaurante;
	
	private BigDecimal preco;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getDescricao() {
		return descricao;
	}

	public void setDescricao(String descricao) {
		this.descricao = descricao;
	}

	public Restaurante getRestaurante() {
		return restaurante;
	}

	public void setRestaurante(Restaurante restaurante) {
		this.restaurante = restaurante;
	}

	public BigDecimal getPreco() {
		return preco;
	}

	public void setPreco(BigDecimal preco) {
		this.preco = preco;
	}

}
