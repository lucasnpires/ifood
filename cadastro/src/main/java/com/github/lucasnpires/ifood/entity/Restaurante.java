package com.github.lucasnpires.ifood.entity;

import java.util.Date;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;

@Entity
@Table(name = "restaurante")
public class Restaurante extends PanacheEntityBase {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private String proprietario;
	
	private String cnpj;
	
	private String nome;
	
	@OneToOne(cascade = CascadeType.ALL)
	private Localizacao localizacao;
	
	@CreationTimestamp
	private Date dtCriacao;
	
	@UpdateTimestamp
	private Date dtAtualizacao;
	
	public Restaurante() {}
	
	public Restaurante(String proprietario, String nome) {
		this.nome = nome;
		this.proprietario = proprietario;
		
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getProprietario() {
		return proprietario;
	}

	public void setProprietario(String proprietario) {
		this.proprietario = proprietario;
	}

	public String getCnpj() {
		return cnpj;
	}

	public void setCnpj(String cnpj) {
		this.cnpj = cnpj;
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public Localizacao getLocalizacao() {
		return localizacao;
	}

	public void setLocalizacao(Localizacao localizacao) {
		this.localizacao = localizacao;
	}

	public Date getDtCriacao() {
		return dtCriacao;
	}

	public void setDtCriacao(Date dtCriacao) {
		this.dtCriacao = dtCriacao;
	}

	public Date getDtAtualizacao() {
		return dtAtualizacao;
	}

	public void setDtAtualizacao(Date dtAtualizacao) {
		this.dtAtualizacao = dtAtualizacao;
	}
}
