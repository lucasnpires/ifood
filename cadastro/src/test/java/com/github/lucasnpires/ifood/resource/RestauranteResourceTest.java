package com.github.lucasnpires.ifood.resource;

import static io.restassured.RestAssured.given;
import static org.assertj.core.api.Assertions.assertThat;

import javax.ws.rs.core.Response.Status;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.github.lucasnpires.ifood.config.CadastroTestLifecycleManager;
import com.github.lucasnpires.ifood.entity.Restaurante;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.http.ContentType;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(value = CadastroTestLifecycleManager.class)
class RestauranteResourceTest {

	@Test
	@DataSet("restaurantes-cenario-1.yml")
	void testBuscarRestaurantes() {
		String resultado = given().when().get("/restaurantes").then().statusCode(Status.OK.getStatusCode()).extract()
				.asString();

		Approvals.verifyJson(resultado);
	}

	@Test
	void testCadastrarRestaurante() {
		Restaurante restaurante = new Restaurante("ID do Keyclock 3", "Restaurante 3");
		given().contentType(ContentType.JSON).when().body(restaurante).post("/restaurantes").then()
				.statusCode(Status.CREATED.getStatusCode());

		Restaurante findById = Restaurante.findById(1L);

		assertThat(restaurante.getNome()).isEqualTo(findById.getNome());
		assertThat(restaurante.getProprietario()).isEqualTo(findById.getProprietario());
	}

	@Test
	@DataSet("restaurantes-cenario-1.yml")
	void testUpdateRestaurante() {
		Restaurante restaurante = new Restaurante();
		restaurante.setNome("Novo Restaurante 1");

		given().contentType(ContentType.JSON).when().body(restaurante).put("/restaurantes/1").then()
				.statusCode(Status.NO_CONTENT.getStatusCode());

		Restaurante findById = Restaurante.findById(1L);

		assertThat(restaurante.getNome()).isEqualTo(findById.getNome());
	}

	@Test
	void testUpdateRestauranteNotFound() {
		given().contentType(ContentType.JSON).when().body(new Restaurante()).put("/restaurantes/6").then()
				.statusCode(Status.NOT_FOUND.getStatusCode());
	}

	@Test
	@DataSet("restaurantes-cenario-1.yml")
	void testDeleteRestaurante() {
		given().contentType(ContentType.JSON).when().delete("/restaurantes/1").then()
				.statusCode(Status.NO_CONTENT.getStatusCode());
	}

	@Test
	void testDeleteRestauranteNotFound() {
		given().contentType(ContentType.JSON).when().delete("/restaurantes/10").then()
				.statusCode(Status.NOT_FOUND.getStatusCode());
	}
}
