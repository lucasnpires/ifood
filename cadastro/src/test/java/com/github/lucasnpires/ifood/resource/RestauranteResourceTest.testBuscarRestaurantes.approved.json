[
  {
    "id": 1,
    "nome": "Restaurante 1",
    "proprietario": "ID do Keyclock 1"
  },
  {
    "id": 2,
    "nome": "Restaurante 2",
    "proprietario": "ID do Keyclock 2"
  }
]